contracts/ColdStaking.sol

```diff
@@ -0,0 +1,160 @@
+pragma solidity ^0.4.18;
+
+import './safeMath.sol';
+
+contract ColdStaking {
+
+    using SafeMath for uint256;
+
+    event StartStaking(address addr, uint256 value, uint256 weight, uint256 init_block);
+
+    event WithdrawStake(address staker, uint256 weight);
+
+    event Claim(address staker, uint256 reward);
+
+    event FirstStakeDonation(address _address, uint256 value);
+
+
+    struct Staker
+    {
+        uint256 weight;
```

Comment:
changing init_block to uint64 would save gas, but worth time / pain?

Reply:
Weight is denominated in WEI in fact. As the result, lets keep the value type the same as `msg.value` (uint256)